src/giftCards/components/GiftCardStatusChip/GiftCardStatusChip.tsx

```diff
@@ -0,0 +1,61 @@
+import StatusChip from "@saleor/components/StatusChip";
+import { StatusType } from "@saleor/components/StatusChip/types";
+import {
+  ExtendedGiftCard,
+  GiftCardBase
+} from "@saleor/giftCards/GiftCardUpdate/providers/GiftCardDetailsProvider/types";
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { giftCardUpdatePageHeaderMessages as giftCardStatusChipMessages } from "../../GiftCardUpdate/GiftCardUpdatePageHeader/messages";
+
+interface GiftCardStatusChipProps<
+  T extends ExtendedGiftCard<GiftCardBase & { isActive: boolean }>
+> {
+  giftCard: T;
+}
+
+function GiftCardStatusChip<
+  T extends ExtendedGiftCard<GiftCardBase & { isActive: boolean }>
+>({ giftCard }: GiftCardStatusChipProps<T>) {
+  const { isExpired, isActive } = giftCard;
+  const intl = useIntl();
+
+  const getGiftCardChipStatus = () => {
+    if (isExpired) {
```

Comment:
I think in case number of these statuses grow I'd much rather define the data in one place instead of two, so
```suggestion
const getGiftCardChipData = () => {
    if (isExpired) {
      return { 
        status: StatusType.NEUTRAL,
        label: giftCardStatusChipMessages.expiredStatusLabel
      }
      
      ...
      
      return null;
```

And then do 
```suggestion
   const statusChipData = getGiftCardChipData()
   
   if (!data) { return null ;} // we should also take that into consideration
   
    return (
      <StatusChip
        size="md"
        status={data.status}
        label={data.label}
      />
  );
```

Reply:
@bmigirl `StatusChip` returns null when label is null

Reply:
Okay, then we can skip the `return null` here, but I'd still leave it as one function instead of two. Unless there's some benefit here over having one?